scripts/helper_scripts/new-parsers/src/bin/functional-analysis.rs

```diff
@@ -0,0 +1,135 @@
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{BufRead, BufWriter, Write};
+use std::path::PathBuf;
+
+use clap::Parser;
+
+use unipept::utils::files::{open_read, open_write};
+
+fn main() {
+    let args = Cli::parse();
+
+    let reader = open_read(&args.input_file);
+    let mut writer = open_write(&args.output_file);
+
+    let mut current_pept: String = String::new();
+
+    let mut num_prot: u32 = 0;
+    let mut num_annotated_go: u32 = 0;
+    let mut num_annotated_ec: u32 = 0;
+    let mut num_annotated_ip: u32 = 0;
+    let mut done: u64 = 0;
+
+    let mut m: HashMap<String, u32> = HashMap::new();
+
+    for line in reader.lines() {
+        match line {
+            Ok(s) => {
+                let row: Vec<String> = s.split('\t').map(String::from).collect();
```

Comment:
Don't do the `.map(String::from)`, you only have to make a string from `row[0]` where you're currently cloning it.